python/src/tests/tasks/test_differential_expression.py

```diff
@@ -151,54 +162,54 @@ def load_correct_definition(self):
     """
 
     @pytest.fixture
-    def mock_dynamo_get(self):
-        with mock.patch("boto3.resource") as m:
-            mockDynamo = MockDynamoClass()
-            m.return_value = mockDynamo
-            yield (m, mockDynamo)
+    def mock_S3_get(self):
+        with mock.patch("boto3.client") as m:
+            mockS3 = MockS3Class()
+            m.return_value = mockS3
+            yield (m, mockS3)
 
     def test_throws_on_missing_parameters(self):
         with pytest.raises(TypeError):
             DifferentialExpression()
 
     @responses.activate
-    def test_cell_sets_get_queried_appropriately(self, mock_dynamo_get):
-        m, dynamodb = mock_dynamo_get
-        m.return_value = dynamodb
+    def test_cell_sets_get_queried_appropriately(self, mock_S3_get):
+        m, s3 = mock_S3_get
```

Comment:
This (lines 177 and 178) is already done in the fixture. Can be deleted from individual tests. We may need to had something to prevent having a warning about the unused `mock_s3_get` parameter

Reply:
I don't understand @xverges -- could you elaborate?

Reply:
@marcellp, we are using the fixture to setup the mock, but we are not using what is yielded by it for anything in the test bodies.
```
    @pytest.fixture
    def mock_S3_get(self):
        with mock.patch("boto3.client") as m:
            mockS3 = MockS3Class()
            m.return_value = mockS3
            yield (m, mockS3)
```
This code is executed by any test function that has a `mock_S3_get` parameter. Currently, on every test body we are doing again what was already done by the fixture body. 

@cosa65 is changing it and may be about to prove my understanding wrong :-D